### PR TITLE
Exclude ‘Trigger’ step time from build time.

### PR DESCRIPTION
**Pull request by kkateq**
This is an attempt to fix https://jira.hq.unity3d.com/browse/CDS-629.

`Trigger` step's time was excluded from the build time summary.

### CODE REVIEW

File: master/buildbot/status/build.py [hunk 1]
```diff
@@ -629,6 +629,14 @@ def get_artifacts(self):
                 self.artifacts = artifacts
             return self.artifacts
 
+    def get_trigger_steps_time(self):
+        trigger = 0
+        trigger_steps = [step.finished - step.started for step in self.steps if step.finished and str(TriggerType) == step.step_type ]
```
Comment (mariangemarcano) [inline master/buildbot/status/build.py]: @kkateq 
I tried out and it doesn't work with a subclass of the trigger step using 
TriggerType = "<class 'buildbot.steps.trigger.Trigger'>"|

from buildbot.steps.trigger import Trigger

class PartitionTrigger(Trigger):
    name = "PartitionTrigger"

maybe @campos-ddc or @sualitu has a suggestion for this?

Comment (kkateq) [reply master/buildbot/status/build.py]: @mariangemarcano probably I understood you wrongly. I thought we should not include subclasses. That is why it doesn't work of course. Let me redo that.

Comment (kkateq) [reply master/buildbot/status/build.py]: Are you ok with the approach of checking step type? Would it be nicer to have it built in step status or make it configurable? Just a few other thoughts, say if we get to exclude other steps(hypothetically not derived from Trigger), or include this one back, then each time source code has to be patched.

Comment (mariangemarcano) [reply master/buildbot/status/build.py]: > Are you ok with the approach of checking step type?

@kkateq I think I would explore other options for this

**Comment by kkateq** [comment]
@sualitu could you take a look on this PR?

**Review by sualitu — APPROVED** [review]

**Pull request closed without merging**